iggy/src/utils/timestamp.rs

```diff
@@ -1,8 +1,16 @@
-use std::time::{SystemTime, UNIX_EPOCH};
+use chrono::{DateTime, Utc};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
 
 pub fn get() -> u64 {
     SystemTime::now()
         .duration_since(UNIX_EPOCH)
         .unwrap()
         .as_micros() as u64
 }
+
+pub fn to_string(timestamp: u64, format: &str) -> String {
+    let system_time = UNIX_EPOCH + Duration::from_micros(timestamp);
+    let date_time = DateTime::<Utc>::from(system_time);
+
+    date_time.format(format).to_string()
+}
```

Comment:
wrap `SystemTime` in your struct, something like
`struct Timestamp(SystemTime)` and implement relevant methods, like `to_string() -> String`, maybe `to_micros() -> u64`